File: examples/currency/convert/node/convertUsdToGbp.js
const { CurrencyService } = require("m3o/currency");

// Convert returns the currency conversion rate between two pairs e.g USD/GBP
async function convertUsdToGbp() {
  let currencyService = new CurrencyService(process.env.MICRO_API_TOKEN);
  let rsp = await currencyService.convert({
    from: "USD",
    to: "GBP",
  });
  console.log(rsp);
}

convertUsdToGbp();

File: examples/currency/convert/go/convertUsdToGbp.go
package example

import (
	"fmt"
	"github.com/micro/services/clients/go/currency"
	"os"
)

// Convert returns the currency conversion rate between two pairs e.g USD/GBP
func ConvertUsdToGbp() {
	currencyService := currency.NewCurrencyService(os.Getenv("MICRO_API_TOKEN"))
	rsp, err := currencyService.Convert(&currency.ConvertRequest{
		From: "USD",
		To:   "GBP",
	})
	fmt.Println(rsp, err)
}
